### PR TITLE
Feature/Auto-calibrate hood

### PR DESCRIPTION
**Pull request by stephenjust**

### CODE REVIEW

File: src/main/java/competition/IdealElectricalContract.java [hunk 1]
```diff
@@ -32,6 +32,10 @@ public boolean isHoodReady() {
         return true;
     }
 
+    public boolean isHoodLimitSwitchReady() {
+        return false;
```
Comment (JohnGilb) [inline src/main/java/competition/IdealElectricalContract.java]: everything always works "ideally"

File: src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java [hunk 2]
```diff
@@ -33,9 +37,38 @@ public void initialize(){
     }
 
     @Override
-    protected void calibratedMachineControlAction(){
+    protected void calibratedMachineControlAction() {
+        if (!this.hood.isCalibrated()) {
```
Comment (JohnGilb) [inline src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java:41]: This block won't be evaluated - `calibratedMachineControlAction` is only ever called if the system is in a calibrated state.

File: src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java [hunk 3]
```diff
@@ -33,9 +37,38 @@ public void initialize(){
...
+        }
+
         double power = pid.calculate(hood.getGoalPercent(), hood.getPercentExtended());
-        hood.setPower(power);
+        this.hood.setPower(power);
+    }
+
+    @Override
+    protected void uncalibratedMachineControlAction() {
+        if (this.hood.isCalibrated()) {
```
Comment (JohnGilb) [inline src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java:52]: also unnecessary, for similar reasons

**Review by JohnGilb — APPROVED** [review]

File: src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java (hunk 2 above)
Comment (stephenjust) [reply src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java:41]: Will leave for safety ⚠️

File: src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java (hunk 3 above)
Comment (stephenjust) [reply src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java:52]: safety

**Pull request merged**